### PR TITLE
📖 Add release team process

### PR DESCRIPTION
**Pull request by CecileRobertMichon**
**What this PR does / why we need it**: As discussed in office hours and in https://github.com/kubernetes-sigs/cluster-api/issues/6442, this document introduces the concept of a CAPI release team. Note that this document is intended to be a starting point for the release team. It is not a complete release process document.

Most of this proposal comes from the discussions in https://docs.google.com/document/d/1DQIsu2nLvkPD5x4c0GW7oYZjb1PUINv4mGPzB_PmPws/edit#. Kudos to @LappleApple for starting it.

A lot of this process is inspired from https://github.com/kubernetes/sig-release/blob/master/release-team/README.md, credits go to the Kubernetes release SIG and past release teams for making it a great resource.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

### CODE REVIEW

File: docs/developer/release-team.md
Comment (CecileRobertMichon) [inline docs/developer/release-team.md:68]: @sbueringer I'd love your input on this based on your experience with doing release tasks in the past

File: docs/developer/release-team.md
Comment (elmiko) [inline docs/developer/release-team.md:54]: given the language in other parts of this doc, i wonder if we should explicitly note that it's ok for members of release team X to continue for release X+1 if they would like?

**Review by elmiko — COMMENTED** [review]
i think this looks great, thanks for putting it together @CecileRobertMichon  =)

i have a question about the team selection.

File: docs/developer/release-team.md
Comment (joekr) [inline docs/developer/release-team.md:60]: Given the term is a quarter I wonder if we shouldn't have a backup plan for each of the roles. Example: Release lead has to step away for a while (life or work related) who takes over? does the lead find a replacement, do the Comms or CI person take over?

File: docs/developer/release-team.md
Comment (furkatgofurov7) [inline docs/developer/release-team.md:48]: Is "Enhancements Lead" out of the scope for this document for now since it is a starting point for the release team?

File: docs/developer/release-team.md
Comment (furkatgofurov7) [inline docs/developer/release-team.md:36]: I am not well aware of the CAPI process yet and it might be that is already the case, but maybe should also add a note about "Security" releases (for all unplanned or embargoed releases including critical CVEs) or in later iterations of this document if turns out to be necessary? if so, adding the actions and responsible person for that release (AFAIU it should be release lead?) can fit in this list.

**Review by furkatgofurov7 — COMMENTED** [review]
Great write-up, learned a lot about the release team/process from the document 💯

File: docs/developer/release-team.md
Comment (CecileRobertMichon) [reply docs/developer/release-team.md:48]: Yes for now we've decided to keep enhancements (CAEP in CAPI terms) separate for simplicity

File: docs/developer/release-team.md
Comment (CecileRobertMichon) [reply docs/developer/release-team.md:54]: Good point, will add that language

File: docs/developer/release-team.md
Comment (CecileRobertMichon) [reply docs/developer/release-team.md:60]: I think we'd want to have someone take over at that point, maybe the shadow can take over? We can also deal with this on a case by case basis to start and formalize the process as we learn from the first iterations

File: docs/developer/release-team.md
Comment (CecileRobertMichon) [reply docs/developer/release-team.md:36]: That's a good question. AFAIK so far we haven't had to deal with embargoed releases / critical CVEs and I don't know if we have an established process, @vincepri @fabriziopandini @sbueringer thoughts?

**Review by ykakarap — COMMENTED** [review]
Looks good overall 🚀

File: docs/developer/release-team.md
Comment (jeremyrickard) [reply docs/developer/release-team.md:32]: This is really similar to what the release team does for k/k. If you have folks that will consume and validate the beta releases, I think that would be a good thing.

File: docs/developer/release-team.md
Comment (jeremyrickard) [reply docs/developer/release-team.md:60]: We’ve had a few instances of this with the Kubernetes release team. Shadows help, having folks that can step back in alsonis very helpful for these kind of scenarios. Definitely plan for it to happen, lots of things can happen in a quarter.

File: docs/developer/release-team.md
Comment (jeremyrickard) [inline docs/developer/release-team.md:41]: SIG Release has a subproject focused on improving the tooling and it’s mostly composed of folks that held various roles on the RT. Having the experience of doing the releases helps people know what to improve, so +1 to this

File: docs/developer/release-team.md
Comment (jeremyrickard) [reply docs/developer/release-team.md:60]: This is great to think about and plan for. We’ve had a few instances of this happening and it’s worked pretty well to have the shadows step up. One thing that can help this is to make good runbook/handbooks for what needs to happen and when it needs to happen (and update that as you find better ways of working).

File: docs/developer/release-team.md
Comment (fabriziopandini) [inline docs/developer/release-team.md:42]: nit
```suggestion
- Track tasks needed to add support for new Kubernetes versions in the next CAPI release
```

File: docs/developer/release-team.md
Comment (fabriziopandini) [inline docs/developer/release-team.md:66]: Might be in a follow-up iteration, but I would like to see two sections in the paragraph:
- one for individuals willing to contribute, explaining why this could be an interesting and hopefully fun experience
- one for helping people willing to contribute in making the case with their managers, explaining why this is important for companies leveraging in CAPI

**Review by fabriziopandini — COMMENTED** [review]
@CecileRobertMichon thanks for driving this!

File: docs/developer/release-team.md
Comment (fabriziopandini) [reply docs/developer/release-team.md:36]: I also don't remember us dealing with embargoed releases / critical CVEs.
Due to this and the fact that we are just starting with this process, I personally think that for now we can assume to treat everything as a "simple" patch, conscious that the attention to communication and scheduling should be proportional to the severity of the issues we are dealing with.

File: docs/developer/release-team.md
Comment (fabriziopandini) [reply docs/developer/release-team.md:68]: It isn't an easy call, but rolling my dice...
Assuming 0,5md for every week in the release cycle we can round up things to 10md in a quarter.
But considering we are starting I think we should double the number for now...

File: docs/developer/release-team.md
Comment (fabriziopandini) [reply docs/developer/release-team.md:28]: Also, what about aiming at three releases/four months cycle per year for starting?
It will be already a great improvement from what we have now with a little bit less pressure on the team/a smmother transition from the current 2 releases per year we historically managed to create (and it also opens up to align our candence with Kubernetes cadence)

File: docs/developer/release-team.md
Comment (CecileRobertMichon) [reply docs/developer/release-team.md:68]: what is "md"?

File: docs/developer/release-team.md
Comment (CecileRobertMichon) [reply docs/developer/release-team.md:32]: Added both rc tags and cherry picks

File: docs/developer/release-team.md
Comment (CecileRobertMichon) [reply docs/developer/release-team.md:50]: Also part of non-goals

File: docs/developer/release-team.md
Comment (CecileRobertMichon) [reply docs/developer/release-team.md:56]: Added to the duration of term section

File: docs/developer/release-team.md
Comment (CecileRobertMichon) [reply docs/developer/release-team.md:68]: Added a paragraph below

File: docs/developer/release-team.md
Comment (CecileRobertMichon) [inline docs/developer/release-team.md:80]: @fabriziopandini please let me know if you have other ideas :)

File: docs/developer/release-team.md
Comment (CecileRobertMichon) [reply docs/developer/release-team.md]: Changed to four-month cycle, the google doc had quarterly in one place and four months in another

File: docs/developer/release-team.md
Comment (CecileRobertMichon) [reply docs/developer/release-team.md]: I put 4-8 hours to start, we'll probably want to update this and make it more granular as we go

**Comment by CecileRobertMichon** [comment]
Addressed all comments, thanks for all the reviews!

File: docs/developer/release-team.md
Comment (fabriziopandini) [reply docs/developer/release-team.md]: sorry if I did not clarified, 1 md = 8 working hours

**Comment by enxebre** [comment]
It's probably worth to reference this here https://github.com/kubernetes-sigs/cluster-api/blob/main/CONTRIBUTING.md#finding-things-that-need-help
Thanks Cecile!
/lgtm

File: docs/developer/release-team.md
Comment (fabriziopandini) [reply docs/developer/release-team.md:80]: That's a great start, thanks. 
We can add more in follow-up iterations (as well as a "convince your boss" section)

File: docs/developer/release-team.md
Comment (killianmuldoon) [inline docs/developer/release-team.md:36]: nit 
```suggestion
- Create beta and release candidate (rc) tags for upcoming minor releases
```

File: docs/developer/release-team.md
Comment (killianmuldoon) [inline docs/developer/release-team.md:42]: What does maintain the milestone involve here?

File: docs/developer/release-team.md
Comment (killianmuldoon) [inline docs/developer/release-team.md:49]: Does this mean the release team will have a minimum of three members?

**Review by killianmuldoon — COMMENTED** [review]
Looks great overall! Just had a couple of questions and a nit.

File: docs/developer/release-team.md
Comment (sbueringer) [inline docs/developer/release-team.md:37]: No strong opinions, but to ensure this the release teams has to be aware of all merged PRs.

I wonder if that should be the responsibility of PR reviewers and especially approvers instead. I can imagine this would scale better as PR reviewers/approvers are already aware of PRs and their content while release team members don't necessarily watch all PRs (? or maybe they do)

File: docs/developer/release-team.md
Comment (sbueringer) [inline docs/developer/release-team.md:38]: Maybe it's to low-level, do you want to add something about creating release branches?

File: docs/developer/release-team.md
Comment (sbueringer) [inline docs/developer/release-team.md:44]: Should we add something about managing CAPI version <=> Kubernetes version compatibility?

Not sure how to express it, but one example is: v1.25.0 was released last week, it's straightforward that v1.3 will support it, but what about v1.2.x. (consequences of the decision are that we have to cherry-pick certain PRs back into release-1.2 and ensure CI is configured accordingly)

File: docs/developer/release-team.md
Comment (sbueringer) [inline docs/developer/release-team.md:51]:
```suggestion
- **CI Signal/Bug Triage/Automation Manager**: Assumes the responsibility of the quality gate for the release and makes sure blocking issues and bugs are triaged and dealt with in a timely fashion. Helps improve release automation and tools.
```
nit

File: docs/developer/release-team.md
Comment (sbueringer) [reply docs/developer/release-team.md:49]: I think roles are not necessarily mapped 1:1. I.e. I would say at most 1 person per role (+ shadow), but if necessary 1 person can have multiple roles.

But I think we should try to have 3 members + shadows

File: docs/developer/release-team.md
Comment (sbueringer) [inline docs/developer/release-team.md:70]:
```suggestion
As a member of the release team, you should expect to spend approximately 4-8 hours a week on release related activities for the duration of the term. 
```
nit

**Review by sbueringer — COMMENTED** [review]
Thank you very much for driving this. Overall lgtm, just some ideas/suggestsions

File: docs/developer/release-team.md
Comment (CecileRobertMichon) [reply docs/developer/release-team.md:42]: Make milestone applier plugin update PRs, keep the milestone due date and description up to date, etc. I don't think this should include grooming or triaging the actual issues / milestone content if that's your question.

File: docs/developer/release-team.md
Comment (CecileRobertMichon) [reply docs/developer/release-team.md:49]: yeah we're aiming to have 3 people + shadows to start

File: docs/developer/release-team.md
Comment (CecileRobertMichon) [reply docs/developer/release-team.md]: I reworded to "Ensure _only_ eligible PRs are cherry-picked", does that help? I agree that the release team shouldn't be aware of every merge PR but I do think they should be aware of all cherry-pick PRs. That way, PR reviewers can submit a PR for cherry-pick and release team can approve the cherry pick (we don't have a formal process of this at the moment, could be as simple as having release team member lgtm'ing the PR).

File: docs/developer/release-team.md
Comment (CecileRobertMichon) [reply docs/developer/release-team.md:38]: IMO that (and all other manual tasks still required to cut a release) is all implementation details of "Release a minor release every 4 months" above. Besides, I think that's something we can automate in the future instead of having a human do it.

File: docs/developer/release-team.md
Comment (CecileRobertMichon) [reply docs/developer/release-team.md:44]: Let's defer that and all other k8s support tasks for now? as discussed in https://docs.google.com/document/d/1DQIsu2nLvkPD5x4c0GW7oYZjb1PUINv4mGPzB_PmPws/edit?disco=AAAAdQ5HoVE

**Review by ykakarap — COMMENTED** [review]
/lgtm

File: docs/developer/release-team.md
Comment (CecileRobertMichon) [reply docs/developer/release-team.md:38]: https://github.com/kubernetes-sigs/cluster-api/issues/7128

File: docs/developer/release-team.md
Comment (sbueringer) [reply docs/developer/release-team.md]: Sounds good to me!

File: docs/developer/release-team.md
Comment (sbueringer) [reply docs/developer/release-team.md:38]: Sounds good

File: docs/developer/release-team.md
Comment (sbueringer) [reply docs/developer/release-team.md:44]: That's fine!

File: CONTRIBUTING.md
Comment (sbueringer) [inline CONTRIBUTING.md:54]: nit: the link leads to 404 in the book.
I think this can be fixed either by moving the relase-team doc into the book or by changing the link to an absolute link (after merge)

**Comment by sbueringer** [comment]
lgtm pending https://github.com/kubernetes-sigs/cluster-api/pull/7110#discussion_r958366116

**Comment by elmiko** [comment]
thanks for all the hard work everybody, and @CecileRobertMichon for driving this forward. it's generally LGTM to me.

File: CONTRIBUTING.md
Comment (CecileRobertMichon) [reply CONTRIBUTING.md:54]: > I think this can be fixed either by moving the relase-team doc into the book 

even if I move it there is no way to make the link for both the book and the CONTRIBUTING.md file outside the book right? because if it's `developer/release-team.md` in the book that same link won't work in the repo tree.

If so the only way is to change it to an absolute link, which I can do after merge.

Comment (sbueringer) [reply CONTRIBUTING.md:54]: You're right!

**Comment by sbueringer** [comment]
Thx @CecileRobertMichon !

/lgtm

**Review by ykakarap — APPROVED** [review]
Thank you for this @CecileRobertMichon!!

/lgtm

**Comment by fabriziopandini** [comment]
Let's get this rolling! Next step define the first Release team and the next release date!
/lgtm
/approve

File: CONTRIBUTING.md
Comment (schrej) [reply CONTRIBUTING.md:54]: Can we add a custom 404 page to netlify that redirects from `docs/<path>` to `<path>`? Then those links would work on both sites.

**Pull request merged**
